stake-pool/single-pool/src/processor.rs

```diff
@@ -0,0 +1,1078 @@
+
+        invoke_signed(&ix, &[mint, destination, authority], signers)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn token_burn<'a>(
+        validator_vote_key: &Pubkey,
+        token_program: AccountInfo<'a>,
+        burn_account: AccountInfo<'a>,
+        mint: AccountInfo<'a>,
+        authority: AccountInfo<'a>,
+        bump_seed: u8,
+        amount: u64,
+    ) -> Result<(), ProgramError> {
+        let authority_seeds = &[
+            POOL_AUTHORITY_PREFIX,
+            validator_vote_key.as_ref(),
+            &[bump_seed],
+        ];
+        let signers = &[&authority_seeds[..]];
+
+        let ix = spl_token::instruction::burn(
+            token_program.key,
+            burn_account.key,
+            mint.key,
+            authority.key,
+            &[],
+            amount,
+        )?;
+
+        invoke_signed(&ix, &[burn_account, mint, authority], signers)
+    }
+
+    fn process_initialize_mint(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let validator_vote_info = next_account_info(account_info_iter)?;
+        let pool_authority_info = next_account_info(account_info_iter)?;
+        let pool_mint_info = next_account_info(account_info_iter)?;
+        let rent_info = next_account_info(account_info_iter)?;
+        let rent = &Rent::from_account_info(rent_info)?;
+        let system_program_info = next_account_info(account_info_iter)?;
+        let token_program_info = next_account_info(account_info_iter)?;
+
+        let authority_bump_seed = check_pool_authority_address(
+            program_id,
+            validator_vote_info.key,
+            pool_authority_info.key,
+        )?;
+        let mint_bump_seed =
+            check_pool_mint_address(program_id, validator_vote_info.key, pool_mint_info.key)?;
+        check_system_program(system_program_info.key)?;
+        check_token_program(token_program_info.key)?;
+
+        // XXX TODO FIXME validate the vote account so people cant create bullshit pools
+        // maybe put the owner + 1 variant check in a check_vote_account instruction
+
+        let authority_seeds = &[
+            POOL_AUTHORITY_PREFIX,
+            validator_vote_info.key.as_ref(),
+            &[authority_bump_seed],
+        ];
+        let authority_signers = &[&authority_seeds[..]];
+
+        let mint_seeds = &[
+            POOL_MINT_PREFIX,
+            validator_vote_info.key.as_ref(),
+            &[mint_bump_seed],
+        ];
+        let mint_signers = &[&mint_seeds[..]];
+
+        // create the pool mint. user has already transferred in rent
+        let mint_space = spl_token::state::Mint::LEN;
+        let mint_rent = rent.minimum_balance(mint_space);
+
+        // TODO remove this after confirming for myself that spl-token doesnt allow creating rentpigs
+        if pool_mint_info.lamports() < mint_rent {
+            return Err(SinglePoolError::WrongRentAmount.into());
+        }
```

Comment:
token checks that it's rent-exempt https://github.com/solana-labs/solana-program-library/blob/5359039e15460c2f2130d2f6a1450339ae849a56/token/program/src/processor.rs#L48